### PR TITLE
feat(api-v3): allow sending client ID when renewing access token FS-1168

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1168" title="FS-1168" target="_blank">Sub-taskFS-1168</a>  [iOS] Update `POST /access` to API V3
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

in this PR we add the ability to send a client ID as a query parameter to the `/access` token in order to associate the id with a client's session.

#### Test Coverage 

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FS-1168]: https://wearezeta.atlassian.net/browse/FS-1168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SQPIT-764]: https://wearezeta.atlassian.net/browse/SQPIT-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SQPIT-764]: https://wearezeta.atlassian.net/browse/SQPIT-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

File: Source/Public/ZMTransportSession.h [hunk 1]
```diff
@@ -130,6 +123,21 @@ extern NSString * const ZMTransportSessionNewRequestAvailableNotification;
 
 @end
 
+@interface ZMTransportSession (AccessToken)
+
+@property (nonatomic, readonly, nullable) ZMAccessToken *accessToken;
+
+/// Sets the access token failure callback. This can be called only before the first request is fired
+- (void)setAccessTokenRenewalFailureHandler:(ZMCompletionHandlerBlock)handler NS_SWIFT_NAME(setAccessTokenRenewalFailureHandler(handler:));
```
Comment (netbe) [inline Source/Public/ZMTransportSession.h:131]: will the `handler` parameter visible in swift ? 
```suggestion
- (void)setAccessTokenRenewalFailureHandler:(ZMCompletionHandlerBlock)handler NS_SWIFT_NAME(setAccessTokenRenewalFailureHandler(_:));
```
as you have twice handler maybe remove the param or the end of the method

File: Source/Public/ZMTransportSession.h [hunk 2]
```diff
@@ -130,6 +123,21 @@ extern NSString * const ZMTransportSessionNewRequestAvailableNotification;
...
+
+/// Sets the access token success callback
+- (void)setAccessTokenRenewalSuccessHandler:(ZMAccessTokenHandlerBlock)handler NS_SWIFT_NAME(setAccessTokenRenewalSuccessHandler(handler:));
```
Comment (netbe) [inline Source/Public/ZMTransportSession.h:134]: same here

**Review by netbe — APPROVED** [review]
LGTM; just a thought about naming

**Comment by David-Henner** [comment]
release major

**Comment by zenkins** [comment]
Hello, @David-Henner! Merging PR for command **release major** for PR #**261**. Will create a **major** release after merged.
Next version is 93.0.0.

**Pull request merged**